gui/CMakeLists.txt

```diff
@@ -9,7 +9,11 @@ if (BUILD_GUI)
         add_definitions(-DQT_DEBUG)
     endif()
 
-    include_directories(${PROJECT_SOURCE_DIR}/lib/)
+    find_package(Qt5 COMPONENTS Core Widgets PrintSupport REQUIRED)
+    include_directories("${PROJECT_SOURCE_DIR}/lib/"
+                        ${Qt5Widgets_INCLUDE_DIRS}
+                        ${Qt5Core_INCLUDE_DIRS}
+                        ${Qt5PrintSupport_INCLUDE_DIRS})
     include_directories(SYSTEM ${PROJECT_SOURCE_DIR}/externals/tinyxml/)
```

Comment:
it seems to me that it would be better to include the Qt paths here instead, where the SYSTEM paths are defined.

Reply:
Do you insist on the addition of the parameter “[SYSTEM](https://cmake.org/cmake/help/v3.13/command/include_directories.html "Description for include_directories()")” for includes of Qt software components?

Reply:
Yes. I am no CMake expert but it seems reasonable. I would like that the Qt include paths are added as system include paths in the compilation.

Reply:
How do you think about to [make this build parameter configurable](https://cmake.org/cmake/help/v3.13/command/option.html "Description for option()")?

Reply:
Why would somebody turn it off?

Reply:
I don't feel I have time to learn all pecularities about compiling Qt applications with cmake. Is there some best practice?

Reply:
I am trying to clarify the conditional application of the build parameter “SYSTEM” generally.

Reply:
The parameter “SYSTEM” can become the default setting for all included software components in your build configuration.
Would you like to offer other build configurators the choice for omitting this parameter in calls of CMake commands?

Reply:
I find that it will occasionally matter if complete build dependency checks will be performed (or not by the requested default).